src/frontend/planner_test/tests/testdata/output/explain_dot_format.yaml

```diff
@@ -437,3 +437,13 @@
rsion: 1 }], pk: [ColumnOrder { column_index: 1, order_type: Some(OrderType { direction: Ascending, nulls_are: Largest }) }], dist_key_in_pk_indices: [0], value_indices: [0, 1], read_prefix_len_hint: 1, versioned: true, stream_key: [1], vnode_col_idx_in_pk: None, retention_seconds: None, maybe_vnode_count: Some(256) }), rate_limit: None, snapshot_read_barrier_interval: 0, arrangement_table: Some(Table { id: 4, schema_id: 4294967294, database_id: 4294967294, name: \"t\", columns: [ColumnCatalog { column_desc: Some(ColumnDesc { column_type: INT32, column_id: 1, name: \"v1\", version: 1 }), is_hidden: false }, ColumnCatalog { column_desc: Some(ColumnDesc { column_type: SERIAL, column_id: 0, name: \"_row_id\", version: 1 }), is_hidden: true }], pk: [ColumnOrder { column_index: 1, order_type: Some(OrderType { direction: Ascending, nulls_are: Largest }) }], dependent_relations: [], table_type: Table, distribution_key: [1], stream_key: [1], append_only: false, owner: 1, fragment_id: 4294967294, vnode_col_index: None, row_id_index: Some(1), value_indices: [0, 1], definition: \"CREATE TABLE t (v1 INT)\", handle_pk_conflict_behavior: Overwrite, read_prefix_len_hint: 1, watermark_indices: [], dist_key_in_pk: [], dml_fragment_id: None, cardinality: Some(Cardinality { lo: 0, hi: None }), initialized_at_epoch: None, created_at_epoch: None, cleaned_by_watermark: false, stream_job_status: Created, create_type: Foreground, description: None, incoming_sinks: [], initialized_at_cluster_version: None, created_at_cluster_version: None, retention_seconds: None, version_column_index: None, cdc_table_id: None, maybe_vnode_count: Some(256), version: Some(TableVersion { version: 0, next_column_id: 2 }), optional_associated_source_id: None }) })) }], stream_key: [1], append_only: false, identity: \"StreamProject { exprs: [t.v1::Float64 as $expr1, t._row_id] }\", fields: [Field { data_type: Some(DOUBLE), name: \"$expr1\" }, Field { data_type: Some(SERIAL), name: \"t._row_id\" }], node_body: Some(Project(ProjectNode { select_list: [ExprNode { function_type: Cast, return_type: Some(DOUBLE), rex_node: Some(FuncCall(FunctionCall { children: [ExprNode { function_type: Unspecified, return_type: Some(INT32), rex_node: Some(InputRef(0)) }] })) }, ExprNode { function_type: Unspecified, return_type: Some(SERIAL), rex_node: Some(InputRef(1)) }], watermark_input_cols: [], watermark_output_cols: [], nondecreasing_exprs: [], noop_update_hint: false })) }], stream_key: [1], append_only: false, identity: \"StreamLocalApproxPercentile { percentile_col: $expr1, quantile: 0.5:Float64, relative_error: 0.01:Float64 }\", fields: [Field { data_type: Some(INT16), name: \"sign\" }, Field { data_type: Some(INT32), name: \"bucket_id\" }, Field { data_type: Some(INT32), name: \"count\" }], node_body: Some(LocalApproxPercentile(LocalApproxPercentileNode { base: 1.02020202020202, percentile_index: 0 })) }), fragment_type_mask: 16, requires_singleton: false, table_ids_cnt: 0, upstream_table_ids: [4] }" ]
```

Comment:
This is huge, we probably should reduce the fields or chunk this into multilines in futures

Reply:
This is the normal explain output. I don't think it should be so long. Could you investigate, let me know if you are blocked and can't find the root cause.
```
    Fragment 0
    StreamMaterialize { columns: [approx_percentile], stream_key: [], pk_columns: [], pk_conflict: NoCheck }
    ├── tables: [ Materialize: 4294967294 ]
    └── StreamGlobalApproxPercentile { quantile: 0.5:Float64, relative_error: 0.01:Float64 }
        ├── tables: [ GlobalApproxPercentileBucketState: 0, GlobalApproxPercentileCountState: 1 ]
        └── StreamExchange Single from 1

    Fragment 1
    StreamLocalApproxPercentile { percentile_col: $expr1, quantile: 0.5:Float64, relative_error: 0.01:Float64 }
    └── StreamProject { exprs: [t.v1::Float64 as $expr1, t._row_id] }
        └── StreamTableScan { table: t, columns: [v1, _row_id] } { tables: [ StreamScan: 2 ] }
            ├── Upstream
            └── BatchPlanNode

    Table 0
    ├── columns: [ sign, bucket_id, count, _rw_timestamp ]
    ├── primary key: [ $0 ASC, $1 ASC ]
    ├── value indices: [ 0, 1, 2 ]
    ├── distribution key: []
    └── read pk prefix len hint: 0

    Table 1
    ├── columns: [ total_count, _rw_timestamp ]
    ├── primary key: []
    ├── value indices: [ 0 ]
    ├── distribution key: []
    └── read pk prefix len hint: 0

    Table 2
    ├── columns: [ vnode, _row_id, backfill_finished, row_count, _rw_timestamp ]
    ├── primary key: [ $0 ASC ]
    ├── value indices: [ 1, 2, 3 ]
    ├── distribution key: [ 0 ]
    ├── read pk prefix len hint: 1
    └── vnode column idx: 0

    Table 4294967294
    ├── columns: [ approx_percentile, _rw_timestamp ]
    ├── primary key: []
    ├── value indices: [ 0 ]
    ├── distribution key: []
    └── read pk prefix len hint: 0


```

Reply:
yea i just put the display of StreamFragment into a label, within StreamFragment there are stream nodes and node_body which can reference back to a stream node that where things are verbose. I should probably merge the dot logic into StreamGraphFormatter.

The original output doesn't look connected, do you want me to output tables as individual nodes on the graph?

Reply:
>  I should probably merge the dot logic into StreamGraphFormatter.
Yes, let's do that. You can use [this](https://dreampuf.github.io/GraphvizOnline/#digraph%20G%20%7B%0A%0A%20%20subgraph%20cluster_0%20%7B%0A%20%20%20%20style%3Dfilled%3B%0A%20%20%20%20color%3Dlightgrey%3B%0A%20%20%20%20node%20%5Bstyle%3Dfilled%2Ccolor%3Dwhite%5D%3B%0A%20%20%20%20a0%20-%3E%20a1%20-%3E%20a2%20-%3E%20a3%3B%0A%20%20%20%20label%20%3D%20%22process%20%231%22%3B%0A%20%20%7D%0A%0A%20%20subgraph%20cluster_1%20%7B%0A%20%20%20%20node%20%5Bstyle%3Dfilled%5D%3B%0A%20%20%20%20b0%20-%3E%20b1%20-%3E%20b2%20-%3E%20b3%3B%0A%20%20%20%20label%20%3D%20%22process%20%232%22%3B%0A%20%20%20%20color%3Dblue%0A%20%20%7D%0A%20%20start%20-%3E%20a0%3B%0A%20%20start%20-%3E%20b0%3B%0A%20%20a1%20-%3E%20b3%3B%0A%20%20b2%20-%3E%20a3%3B%0A%20%20a3%20-%3E%20a0%3B%0A%20%20a3%20-%3E%20end%3B%0A%20%20b3%20-%3E%20end%3B%0A%0A%20%20start%20%5Bshape%3DMdiamond%5D%3B%0A%20%20end%20%5Bshape%3DMsquare%5D%3B%0A%7D) to render your graph and see if the output is reasonable.

> The original output doesn't look connected, do you want me to output tables as individual nodes on the graph?
I think that's an acceptable approach, given that `StreamGraphFormatter` keeps these as a separate field, and not part of the stream graph.

Reply:
Thanks, the new update integrate the petagraph from StreamGraphFormatter the output is much nicer for stream. I think batch distsql calls the display and need help to review carefully.

Reply:
>  I think batch distsql calls the display and need help to review carefully.

Could you try adding it as an e2e test instead? Inside `e2e_test/batch/explain.slt`.